crates/ironrdp-dvc-pipe-proxy/src/windows/handle.rs

```diff
@@ -0,0 +1,53 @@
+use windows::Win32::Foundation::{CloseHandle, HANDLE};
+
+use crate::windows::WindowsError;
+
+/// A wrapper around a Windows [`HANDLE`].
+///
+/// Whenever possible, you should use [`BorrowedHandle`] or [`OwnedHandle`] instead.
+/// Those are safer to use.
+#[derive(Debug, Clone)]
+pub(crate) struct Handle {
+    raw: HANDLE,
+}
+
+// SAFETY: A `HANDLE` is, by definition, thread safe.
+unsafe impl Send for Handle {}
+
+// SAFETY: A `HANDLE` is simply an integer, no dereferencing is done.
+unsafe impl Sync for Handle {}

```

Comment:
issue: Yes and no. Pointers are just integers as well, but what matters is how they are used. Both objects are pointing to a ressource, and if the handle/pointer is passed to a function, if the operation on the resource is not thread-safe, then this instance of the Handle is not thread safe. My understanding here is that you can only know if its thread safe in the upper layer abstraction, where all the usages are known. It’s not okay to implement a generic handle wrapper with `Send` and `Sync` implemented on it.